### PR TITLE
Add some types in `ace.d.ts`

### PR DESCRIPTION
**Pull request by wafuwafu13**
*Issue #, if available:*

*Description of changes:*
Add some types in `ace.d.ts`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ace.d.ts [hunk 1]
```diff
@@ -20,10 +20,12 @@ export namespace Ace {
     getLine(row: number): string;
     getLines(firstRow: number, lastRow: number): string[];
     getAllLines(): string[];
+    getLength(): number;
```
Comment (wafuwafu13) [inline ace.d.ts:23]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/document.js#L206-L208

File: ace.d.ts [hunk 2]
```diff
@@ -20,10 +20,12 @@ export namespace Ace {
...
     getTextRange(range: Range): string;
     getLinesForRange(range: Range): string[];
     insert(position: Point, text: string): Point;
     insertInLine(position: Point, text: string): Point;
+    insertNewLine(position: Point): Point;
```
Comment (wafuwafu13) [inline ace.d.ts:28]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/document.js#L251-L254

File: ace.d.ts [hunk 3]
```diff
@@ -398,6 +400,8 @@ export namespace Ace {
   export interface EditSession extends EventEmitter, OptionsProvider, Folding {
     selection: Selection;
 
+    // TODO: define BackgroundTokenizer
```
Comment (wafuwafu13) [inline ace.d.ts:403]: https://github.com/ajaxorg/ace/blob/master/lib/ace/background_tokenizer.js

File: ace.d.ts [hunk 4]
```diff
@@ -519,6 +523,8 @@ export namespace Ace {
     removeKeyboardHandler(handler: KeyboardHandler): boolean;
     getKeyboardHandler(): KeyboardHandler;
     getStatusText(): string;
+    onCommandKey(e: any, hashId: number, keyCode: number): boolean;
```
Comment (wafuwafu13) [inline ace.d.ts:526]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/keybinding.js#L140-L143

File: ace.d.ts [hunk 5]
```diff
@@ -519,6 +523,8 @@ export namespace Ace {
...
+    onTextInput(text: string): boolean;
```
Comment (wafuwafu13) [inline ace.d.ts:527]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/keybinding.js#L145-L147

File: ace.d.ts [hunk 6]
```diff
@@ -555,11 +555,17 @@ export namespace Ace {
     toggleRecording(editor: Editor): void;
     replay(editor: Editor): void;
     addCommand(command: Command): void;
-    addCommands(command: Command): void;
+    addCommands(command: Command[]): void;
     removeCommand(command: Command | string, keepCommand?: boolean): void;
+    removeCommands(command: Command[])
```
Comment (wafuwafu13) [inline ace.d.ts:560]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/hash_handler.js#L175-L179

File: ace.d.ts [hunk 7]
```diff
@@ -555,11 +555,17 @@ export namespace Ace {
...
     bindKey(key: string | { mac?: string, win?: string },
       command: CommandLike,
       position?: number): void;
+    bindKeys(keys: {[s: string]: Function}): void;
```
Comment (wafuwafu13) [inline ace.d.ts:564]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/hash_handler.js#L181-L185

File: ace.d.ts [hunk 8]
```diff
@@ -555,11 +555,17 @@ export namespace Ace {
...
+    parseKeys(keyPart: string): {key: string, hashId: number};
```
Comment (wafuwafu13) [inline ace.d.ts:565]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/hash_handler.js#L193

File: ace.d.ts [hunk 9]
```diff
@@ -555,11 +555,17 @@ export namespace Ace {
...
+    findKeyCommand(hashId: number, keyString: string): string | undefined;
```
Comment (wafuwafu13) [inline ace.d.ts:566]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/hash_handler.js#L218-L221

File: ace.d.ts [hunk 10]
```diff
@@ -555,11 +555,17 @@ export namespace Ace {
...
+    handleKeyboard(data: {}, hashId: number, keyString: string, keyCode: string | number): void | {command: string};
```
Comment (wafuwafu13) [inline ace.d.ts:567]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/hash_handler.js#L223

File: ace.d.ts [hunk 11]
```diff
@@ -555,11 +555,17 @@ export namespace Ace {
...
+    getStatusText(editor?: Editor, data: {}): string;
```
Comment (wafuwafu13) [inline ace.d.ts:568]: https://github.com/ajaxorg/ace/blob/e9f53ddf3c789f0042d83a0c9c3163021d028159/lib/ace/keyboard/hash_handler.js#L248-L250

**Comment by nightwing** [comment]
Thank you!

**Pull request merged**